jvb/src/main/java/org/jitsi/videobridge/Conference.java

```diff
@@ -493,10 +493,23 @@ private void maybeSendKeyframeRequest(AbstractEndpoint dominantSpeaker)
         boolean anyEndpointInStageView = false;
         for (Endpoint otherEndpoint : getLocalEndpoints())
         {
-            if (otherEndpoint != dominantSpeaker && otherEndpoint.isInStageView())
+            if (otherEndpoint != dominantSpeaker)
             {
-                anyEndpointInStageView = true;
-                break;
+                // Do not anticipate a switch if all on-stage sources are DESKTOP
+                List<String> onStageSourcesNames = otherEndpoint.getOnStageSources();
+                for (String onStageSourceName : onStageSourcesNames)
+                {
+                    AbstractEndpoint owner = findSourceOwner(onStageSourceName);
```

Comment:
`findSourceOwner` also iterates over all the endpoints, so this is now quadratic in the number of endpoints, which I don't love.  Is there a better way to structure this?